src/current/v24.2/troubleshoot-lock-contention.md

```diff
@@ -0,0 +1,378 @@
+~~~
+
+### Terminal 3
+
+In a third terminal, open another SQL shell to the demo cluster using the `cli` command from the `\demo ls` output:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cockroach sql --certs-dir=/Users/myuser/.cockroach-demo -u demo -d defaultdb
+~~~
+
+In the third SQL shell, set `application_name`:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'Transaction 3';  -- to distinguish between transactions
+~~~
+
+### DB Console
+
+In a web browser, open the DB Console to the demo cluster using the `webui` address from the `\demo ls` output:
+
+~~~
+http://127.0.0.1:8080/demologin?password=demo28806&username=demo
+~~~
+
+## Understand lock contention
+
+### Initial Data
+
+In any of the SQL shells, create a table and insert some data:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+DROP TABLE IF EXISTS t;
+CREATE TABLE t (k INT PRIMARY KEY, v INT);
+INSERT INTO t VALUES (1,1), (2,2), (3,3);
+~~~
+
+### Example 1
+
+In this example, Transaction 1 is a write that blocks Transaction 2, a read, and Transaction 3, a write. Transaction 1 locks key `k=2`. When Transaction 2 tries to read key `k=2`, it experiences lock contention and waits for the lock on the key to be released. Similarly, when Transaction 3 tries to write to key `k=2`, it experiences lock contention and waits for the lock on the key to be released.
+
+Transaction 1 (blocking write)   | Transaction 2 (waiting read) | Transaction 3 (waiting write)
+---------------------------------|------------------------------|--------------------------------
+`BEGIN;`                         |                              |
+`UPDATE t SET v=2012 WHERE k=2;` | `BEGIN;`                     |
+lock k=2                         | `SELECT * FROM t WHERE k=2;` | `BEGIN;`
+                                 | waiting                      | `UPDATE t SET v=2032 WHERE k=2;`
+                                 |                              | waiting
+`COMMIT;`                        | unblocked to proceed         | unblocked to proceed
+success, k=2,v=2012              | k=2, v=2012                  |
+                                 | `COMMIT;`                    | `COMMIT;`
+                                 | success                      | success                                 
+                                 |                              | `SELECT * FROM t WHERE k=2;`
+                                 |                              | k=2, v=2032
+
+### SQL statements
+
+To reproduce Example 1 in CockroachDB in preparation for the next section on how to [identify waiting and blocking transactions](#identify-waiting-and-blocking-transactions) using the Insights page of the Cloud Console or the DB Console, execute the following SQL statements in the given order in the specified terminal.
+
+**Terminal 1**
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET application_name = 'Transaction 4';  -- to distinguish between transactions
```

Comment:
If they're reproducing Example 1 above, shouldn't they keep the Transactions labeled as 1, 2, and 3? It looks like a bit further down we keep Transaction 1, 2, and 3 so this might just be a typo.